eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/annotation/load/VariantAnnotationMongoItemWriter.java

```diff
@@ -0,0 +1,168 @@
+ *
+ * Will be:
+ *
+ * "annot" : {
+ *  "ct" : [
+ *      { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000382410", "codon" : "-", "strand" : "+", "bt" : "protein_coding", "aaChange" : "-", "so" : [ 1631 ] },
+ *      { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000608838", "codon" : "-", "strand" : "+", "bt" : "processed_transcript", "aaChange" : "-",
+ *  "so" : [ 1631 ] } ],
+ *  "xrefs" : [
+ *      { "id" : "DEFB125", "src" : "HGNC" },
+ *      { "id" : "ENST00000382410", "src" : "ensemblTranscript" },
+ *      { "id" : "ENST00000608838", "src" : "ensemblTranscript" },
+ *      { "id" : "ENSG00000178591", "src" : "ensemblGene"
+ */
+public class VariantAnnotationMongoItemWriter extends MongoItemWriter<VariantAnnotation> {
+    private static final Logger logger = LoggerFactory.getLogger(VariantAnnotationMongoItemWriter.class);
+
+    private MongoOperations mongoOperations;
+    private String collection;
+
+    public VariantAnnotationMongoItemWriter(MongoOperations mongoOperations) {
+        this.mongoOperations = mongoOperations;
+    }
+
+    @Override
+    public void setCollection(String collection) {
+        super.setCollection(collection);
+        this.collection = collection;
+    }
+/*    public VariantAnnotationMongoItemWriter() {
+        this.template = mongoTemplate();
+    }
+
+    public MongoTemplate mongoTemplate() {
+        MongoTemplate mongoTemplate;
+        try {
+            //mongoTemplate = new MongoTemplate(new MongoClient(), pipelineOptions.getString(VariantStorageManager.DB_NAME));
+            mongoTemplate = new MongoTemplate(new MongoClient(), "variants");
+        } catch (UnknownHostException e) {
+            throw new RuntimeException("Unable to initialize MongoDB", e);
+        }
+        return mongoTemplate;
+    }*/
+
+    @Override
+    protected void doWrite(List<? extends VariantAnnotation> variantAnnotations) {
+        //// TODO: 30/06/2016 move in constructor?
+        DBObjectToVariantAnnotationConverter converter = new DBObjectToVariantAnnotationConverter();
+
+        for (VariantAnnotation variantAnnotation : variantAnnotations) {
+            //logger.debug("Writing into mongo {}", variantAnnotation);
+
+            String storageId = buildStorageId(variantAnnotation.getChromosome(), variantAnnotation.getStart(),
+                    variantAnnotation.getReferenceAllele(), variantAnnotation.getAlternativeAllele());
+
+            BasicDBObject find = new BasicDBObject("_id", storageId);
+
+            //find existing variant with same Id, and bring only the annotation
+            DBObject existingVariantStorage = mongoOperations.getCollection(collection)
+                    .findOne(find, new BasicDBObject("annot", 1));
+
+            //update annotation in existing variant
+            if (null != existingVariantStorage){
```

Comment:
OK will swap.. I was just following this:
https://en.wikipedia.org/wiki/Yoda_conditions
http://stackoverflow.com/questions/271561/why-does-one-often-see-null-variable-instead-of-variable-null-in-c